kvbc/src/Replica.cpp

```diff
@@ -323,8 +323,25 @@ void Replica::createReplicaAndSyncState() {
   handleNewEpochEvent();
   handleWedgeEvent();
   bftEngine::impl::DbCheckpointManager::instance().initializeDbCheckpointManager(
-      m_dbSet.dataDBClient, m_replicaPtr->persistentStorage(), aggregator_, [this]() -> uint64_t {
-        return getLastBlockId();
+      m_dbSet.dataDBClient,
+      m_replicaPtr->persistentStorage(),
+      aggregator_,
+      [this]() -> uint64_t { return getLastBlockId(); },
+      [](BlockId blockIdAtCheckpoint, const std::string &path) {
+        if (blockIdAtCheckpoint < INITIAL_GENESIS_BLOCK_ID) {
```

Comment:
I am not sure if understood this. we call prepareCheckpointCb_ with lastReachableBlock in the blockchain. So, in which case, we may hit this condition? won't this if condition always  be false. may be i am missing something

Reply:
INITIAL_GENESIS_BLOCK_ID is a constant that is currently set to 1. The check handles a case where a checkpoint is created at block 0 (if done manually). Does that make sense?